how_tos/how_to_add_dynamical_decoupling.ipynb

```diff
@@ -0,0 +1,442 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "fd5c7e76-a9cb-430e-9dc9-aba93e50424b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(\"data/graph_hardware_native_127nodes_001.json\", \"r\") as fin:\n",
+    "    data = json.load(fin)\n",
+    "\n",
+    "cost_operator = SparsePauliOp.from_list(data[\"paulis\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "ed94423b-2e51-4cda-97c8-de587b6119f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "provider = IBMProvider(instance=\"quantum-demonstrations/main/locc\")\n",
+    "backend = provider.get_backend(\"ibm_kyiv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "312e8cad-a9c8-4ad4-a3df-21073859ba3c",
+   "metadata": {},
+   "source": [
+    "## Circuit creation\n",
+    "\n",
+    "The problem is hardware native. Therefor, if we transpile its QAOA circuit to a 127 qubit backend with CNOT gates and a heavy-hexagonal coupling map then we expect to have (i) 288 CX gates per QAOA layer because a 127 qubit backend has 144 edges in its coupling map and each `RZZGate` requires two CNOT gates, and (ii) a CNOT depth of 6 because the heavy-hex coupling map has a maximum node degree of three. We will first construct a circuit with the standard Qiskit transpilation flow and show that it is sub-optimal."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "bb5c892a-beba-4e7e-9de2-1a7e7c6939a0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p = 2\n",
+    "qaoa_circuit = QAOAAnsatz(cost_operator, reps=p)\n",
+    "qaoa_circuit = transpile(qaoa_circuit, backend, optimization_level=3)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "1447445f-594d-4f7b-92b3-543ebf1a8347",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of two-qubit gates: 576\n",
+      "CNOT gate depth: 80\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Number of two-qubit gates:\", qaoa_circuit.count_ops()[\"cx\"])\n",
+    "print(\"CNOT gate depth:\", qaoa_circuit.depth(filter_function=lambda x: x.operation == CXGate()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3a511837-4aa3-4e93-8a2d-ff65d857e9d0",
+   "metadata": {},
+   "source": [
+    "We can see above that the Qiskit transpiler has produced a hardware native circuit, because the number of two-qubit gates is the minimum number we need, i.e., `p * 2 * 144`. However, something went wrong in the circuit depth. We have a CNOT gate depth of 80 when we were expecting `2 * p * 3`. This shows that the gates have been staggereg instead of placed in parallel. To fix this we will transpile with an empty swap network. We chose an empty network because we do not need swaps but we still need the ability of this transpiler pass to place commuting two-qubit gates in parallel."
```

Comment:
```suggestion
    "We can see above that the Qiskit transpiler has produced a hardware native circuit, because the number of two-qubit gates is the minimum number we need, i.e., `p * 2 * 144`. However, something went wrong in the circuit depth. We have a CNOT gate depth of 80 when we were expecting `2 * p * 3`. This shows that the gates have been staggered instead of placed in parallel. To fix this we will transpile with an empty swap network. We chose an empty network because we do not need swaps but we still need the ability of this transpiler pass to place commuting two-qubit gates in parallel."
```